### PR TITLE
Update spring-boot's version to 2.3.4.RELEASE

### PR DESCRIPTION
**Pull request by chenrujun**
- Update `spring-boot` and `spring-cloud`'s version to latest released version. 
 - Update `external dependencies`' version according to [spring-boot-dependencies](https://repo.maven.apache.org/maven2/org/springframework/boot/spring-boot-dependencies/2.3.4.RELEASE/spring-boot-dependencies-2.3.4.RELEASE.pom/) and [spring-cloud-dependencies](https://repo.maven.apache.org/maven2/org/springframework/cloud/spring-cloud-dependencies/Hoxton.SR8/spring-cloud-dependencies-Hoxton.SR8.pom)

### CODE REVIEW

File: eng/versioning/external_dependencies.txt
Comment (alzimmermsft) [inline eng/versioning/external_dependencies.txt:47]: Should this version change to `2.0.34.Final`?

File: sdk/storage/azure-storage-blob-batch/pom.xml
Comment (joshfree) [inline sdk/storage/azure-storage-blob-batch/pom.xml:96]: even though this is just updating external_dependency, please get an acknowledgement from the storage team before merging this;

/cc @rickle-msft @gapra-msft

**Review by joshfree — APPROVED** [review]

File: eng/versioning/external_dependencies.txt
Comment (joshfree) [reply eng/versioning/external_dependencies.txt:47]: good catch @alzimmermsft

Comment (chenrujun) [reply eng/versioning/external_dependencies.txt:47]: @alzimmermsft , 
Thank you for your check.
Currently, `netty-tcnative` is not managed by `spring-boot-dependencies`.
I created a PR: https://github.com/spring-projects/spring-boot/pull/23502

And this time, I'll update `netty-tcnative`'s version to `2.0.34.Final` manually.

**Review by gapra-msft — APPROVED** [review]
Storage changes look good to me

**Review by alzimmermsft — APPROVED** [review]

**Comment by kushagraThapar** [comment]
@chenrujun  I investigated this issue today, and seems like Azure Cosmos SDK has an issue with `netty library version 4.1.52.Final`

This issue is easily reproducible with `ChangeFeedProcessorTest` on this PR. 
These are the error logs I am seeing locally.

```
2020-09-29 17:36:58,661       [main] INFO  com.azure.cosmos.TestNGLogListener - beforeInvocation: TestSuiteBase#beforeSuite
2020-09-29 17:36:58,669       [TestNG-method=beforeSuite-1] INFO  TestSuiteBase - beforeSuite Started
2020-09-29 17:36:58,747       [TestNG-method=beforeSuite-1] INFO  com.azure.cosmos.implementation.RxDocumentClientImpl - Initializing DocumentClient with serviceEndpoint [https://paged-flux-test-account.documents.azure.com:443/], connectionPolicy [ConnectionPolicy{requestTimeout=PT5S, connectionMode=GATEWAY, maxConnectionPoolSize=1000, idleHttpConnectionTimeout=PT1M, idleTcpConnectionTimeout=null, userAgentSuffix='', throttlingRetryOptions=RetryOptions{maxRetryAttemptsOnThrottledRequests=9, maxRetryWaitTime=PT33H20M}, endpointDiscoveryEnabled=true, preferredRegions=null, multipleWriteRegionsEnabled=true, proxyType=null, inetSocketProxyAddress=null, readRequestsFallbackEnabled=true, connectTimeout=null, idleEndpointTimeout=null, maxConnectionsPerEndpoint=0, maxRequestsPerConnection=0}], consistencyLevel [Session], directModeProtocol [Tcp]
2020-09-29 17:36:58,959       [parallel-1] INFO  com.azure.cosmos.implementation.RxDocumentClientImpl - Getting database account endpoint from https://paged-flux-test-account.documents.azure.com:443/
2020-09-29 17:36:59,156       [reactor-http-nio-1] WARN  io.netty.channel.DefaultChannelPipeline - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.channel.ChannelPipelineException: reactor.netty.resources.PooledConnectionProvider$PooledConnectionAllocator$PooledConnectionInitializer.handlerAdded() has thrown an exception; removed.
	at io.netty.channel.DefaultChannelPipeline.callHandlerAdded0(DefaultChannelPipeline.java:624) [netty-transport-4.1.52.Final.jar:?]
	at io.netty.channel.DefaultChannelPipeline.access$100(DefaultChannelPipeline.java:46) [netty-transport-4.1.52.Final.jar:?]
	at io.netty.channel.DefaultChannelPipeline$PendingHandlerAddedTask.execute(DefaultChannelPipeline.java:1463) [netty-transport-4.1.52.Final.jar:?]
	at io.netty.channel.DefaultChannelPipeline.callHandlerAddedForAllHandlers(DefaultChannelPipeline.java:1115) [netty-transport-4.1.52.Final.jar:?]
	at io.netty.channel.DefaultChannelPipeline.invokeHandlerAddedIfNeeded(DefaultChannelPipeline.java:650) [netty-transport-4.1.52.Final.jar:?]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:502) [netty-transport-4.1.52.Final.jar:?]
	at io.netty.channel.AbstractChannel$AbstractUnsafe.access$200(AbstractChannel.java:417) [netty-transport-4.1.52.Final.jar:?]
	at io.netty.channel.AbstractChannel$AbstractUnsafe$1.run(AbstractChannel.java:474) [netty-transport-4.1.52.Final.jar:?]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164) [netty-common-4.1.52.Final.jar:?]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:472) [netty-common-4.1.52.Final.jar:?]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:500) [netty-transport-4.1.52.Final.jar:?]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989) [netty-common-4.1.52.Final.jar:?]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.52.Final.jar:?]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.52.Final.jar:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.util.NoSuchElementException: reactor.netty.resources.PooledConnectionProvider$PooledConnectionAllocator$PooledConnectionInitializer
	at io.netty.channel.DefaultChannelPipeline.getContextOrDie(DefaultChannelPipeline.java:1082) ~[netty-transport-4.1.52.Final.jar:?]
	at io.netty.channel.DefaultChannelPipeline.remove(DefaultChannelPipeline.java:417) ~[netty-transport-4.1.52.Final.jar:?]
	at reactor.netty.resources.PooledConnectionProvider$PooledConnectionAllocator$PooledConnectionInitializer.handlerAdded(PooledConnectionProvider.java:284) ~[reactor-netty-0.9.12.RELEASE.jar:?]
	at io.netty.channel.AbstractChannelHandlerContext.callHandlerAdded(AbstractChannelHandlerContext.java:938) [netty-transport-4.1.52.Final.jar:?]
	at io.netty.channel.DefaultChannelPipeline.callHandlerAdded0(DefaultChannelPipeline.java:609) [netty-transport-4.1.52.Final.jar:?]
	... 14 more
```

@moderakh  @FabianMeiswinkel  - we will need to debug this issue as this will block us / everyone moving to the latest netty version. 

We can let others move by updating external_dependencies.txt file and keeping netty dependency for cosmos on current versions, but it will be an issue for us soon once we want to upgrade.

**Review by kushagraThapar — DISMISSED** [review]
Please do not merge this as it will break azure-cosmos SDK.

**Review by moderakh — CHANGES_REQUESTED** [review]
this should not get merged till cosmos db issue on newer version of netty is fixed.
Please don't merge this.

**Comment by kushagraThapar** [comment]
@chenrujun  - It was a netty dependency issue in azure-cosmos Java SDK. I have fixed it through this PR - https://github.com/Azure/azure-sdk-for-java/pull/16602

Once my PR is merged, you can rebase on the latest master and re-run the cosmos CI to check everything looks good.

**Comment by kushagraThapar** [comment]
@chenrujun - We have solved this dependency conflict issue here - https://github.com/Azure/azure-sdk-for-java/pull/16604
Please rebase your PR with master and run the tests again to verify if the issue is resolved now.

**Review by yiliuTo — APPROVED** [review]
LGTM

**Pull request merged**
